src/citrine/resources/data_concepts.py

```diff
@@ -439,12 +469,33 @@ def register(self, model: ResourceType, *, dry_run=False):
 
         temp_scope = str(uuid4())
         scope = temp_scope if dry_run else CITRINE_SCOPE
-        GEMDJson(scope=scope).dumps(model)  # This apparent no-op populates uids
+        set_uuids(model, scope=scope)
         dumped_data = replace_objects_with_links(scrub_none(model.dump()))
-        recursive_foreach(model, lambda x: x.uids.pop(temp_scope, None))  # Strip temp uids
 
         data = self.session.post_resource(path, dumped_data, params=params)
-        return self.build(data)
+        registered = self.build(data)
+
+        recursive_foreach(model, lambda x: x.uids.pop(temp_scope, None))  # Strip temp uids
+        if not dry_run:
+            # Platform may add a CITRINE_SCOPE uid and citr_auto tags; update locals
+            model.uids.update({k: v for k, v in registered.uids.items()})
+            if registered.tags is not None:
+                if model.tags is None:  # This is somehow hit by nextgen-devkit tests
+                    model.tags = list()  # pragma: no cover
+                model.tags.extend([tag for tag in registered.tags
+                                   if re.match(f"^{CITRINE_TAG_PREFIX}::", tag)])
+        else:
+            # Remove of the tags/uids the platform spuriously added
+            # this might leave objects with just the temp ids, which we want to strip later
+            if CITRINE_SCOPE not in model.uids:
+                registered.uids.pop(CITRINE_SCOPE, None)
+            if registered.tags is not None:
+                todo = [tag for tag in registered.tags
+                        if re.match(f"^{CITRINE_TAG_PREFIX}::", tag)]
+                for tag in todo:  # Covering this block would require dark art
+                    if tag not in model.tags:
+                        registered.tags.remove(tag)
+        return registered
```

Comment:
@kroenlein any chance you remember the context of this tag manipulation use case? It's fine if "no"... I'm just trying to trace back the `citr_auto` use cases.

Reply:
The platform returns registered versions of all objects during the register_all operation.  However, because of order of operations, if enough materials are passed so that submissions must be batched, all material_runs returned by the platform will be annotated as terminal materials, since they are all registered before the ingredients that would reference them.

Reply:
The SDK still batches into groups of 50 because of service limits.  Is that no longer the case?

We used the full `citr_auto::` rather than specifically terminal checks because we figured it was not a user-supplied value, so it probably should be that user facing.

Reply:
So the SDK adds the appropriate terminal material designation to all objects and then it batches after that?